harness/determined/core/_context.py

```diff
@@ -152,14 +157,21 @@ def init(
         preempt_mode (``core.PreemptMode``, optional): Configure the calling pattern for the
             ``core_context.preempt.should_preempt()`` method.  See
             :class:`~determined.core.PreemptMode` for more detail.  Defaults to ``WorkersAskChief``.
-        storage_manager: Internal use only.
```

Comment:
oh, this was already explicitly documented as internal?  Let's just remove support for it instead of logging a warning.  No user has any right to complain if they were using internal interfaces.  We'll have to fix up some unit tests but that's fine.